ow_plexil/src/plexil-adapter/FaultDependencies.cpp

```diff
@@ -0,0 +1,323 @@
+      }
+  }
+  // If subsystem inoperable flag changes, we need to cascade inoperable fault to impacted subsystems
+  if (prev_inoperable_state != m_subsystems[name].num_inoperable){
+    cascadeFault(name, status);
+  }
+}
+
+void FaultDependencies::cascadeFault(const std::string &subsystem_name, int status){
+  std::unordered_set<std::string> visited;
+  std::stack<std::string> nodes;
+
+  visited.insert(subsystem_name);
+  nodes.push(subsystem_name);
+  
+    // BFS through all subsystem impacts
+    while(!nodes.empty()){
+      std::string current_node = nodes.top();
+      nodes.pop();
+
+      for (auto i: m_subsystems[current_node].impacts){
+        // If it is a procedure we do not need to add it to BFS stack, and can just update the inoperable flag
+        if (i.second == "Procedure"){
+          m_procedures[i.first].num_inoperable = status;
+        }
+        // Otherwise we update subsystem and add it to stack
+        else{
+          if (visited.insert(i.first).second && m_subsystems.find(i.first) != m_subsystems.end()){
+            // increment or decrement num_non_local_faults flag
+            if (status){
+              m_subsystems[i.first].num_non_local_faults += 1;
+              m_subsystems[i.first].num_inoperable += 1;
+            }
+            else{
+              m_subsystems[i.first].num_non_local_faults -= 1;
+              m_subsystems[i.first].num_inoperable -= 1;
+            }
+            nodes.push(i.first);
+          }
+        }
+      }
+    }
+}
+
+void FaultDependencies::DebugPrint() const{
+  // Prints out current fault info for all faults and subsystems
+  ROS_INFO_STREAM("FAULTS: " );
+  ROS_INFO_STREAM("------------------------------------------------------------------------"
+            << "-----------------------------" 
+            );
+  ROS_INFO_STREAM(std::setw(40) << std::left << "NAME" << std::setw(20) << "FAULTY");
+  ROS_INFO_STREAM("------------------------------------------------------------------------"
+            << "-----------------------------" 
+            );
+  for (auto fault: m_faults){
+    ROS_INFO_STREAM(std::left << std::setw(40) << fault.second.name << std::setw(20) << fault.second.faulty );
+  }
+
+  ROS_INFO_STREAM("\n" );
+  ROS_INFO_STREAM("SUBSYSTEMS/PROCEDURES STATUS" );
+  ROS_INFO_STREAM("------------------------------------------------------------------------"
+            << "-----------------------------" 
+            );
+
+  ROS_INFO_STREAM(std::setw(25) << std::left << "NAME" << std::setw(17) << "TYPE" << std::setw(17) << "NON-LOCAL FAULT" 
+  << std::setw(17) << "LOCAL FAULT" << std::setw(17) << "FAULTY" << "INOPERABLE" );
+  ROS_INFO_STREAM("------------------------------------------------------------------------"
+            << "-----------------------------" 
+            );
+  for (auto subsystem: m_subsystems){
+    ROS_INFO_STREAM(std::left << std::setw(25) << subsystem.second.name << std::setw(17) << "Subsystem" << std::setw(17) << 
```

Comment:
Please shorten this and other lines >120 characters to conform to our max 120 character line length.